src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatWriterGenerator.cs

```diff
@@ -286,6 +286,12 @@ private int WriteMembers(ClassDataContract classContract, LocalBuilder extension
                         _ilg.Load(_contextArg);
                         _ilg.Call(XmlFormatGeneratorStatics.StoreIsGetOnlyCollectionMethod);
                     }
+                    else
+                    {
+                        _ilg.Load(_contextArg);
+                        _ilg.Call(XmlFormatGeneratorStatics.ResetIsGetOnlyCollectionMethod);
```

Comment:
The json generators use a different _ilg.Call overload which looks a little cleaner than this. Did you use this version to be consistent with the first statement block of the if?